src/buildings/nobHarborBuilding.cpp

```diff
@@ -110,13 +110,15 @@ nobHarborBuilding::nobHarborBuilding(const MapPoint pos, const unsigned char pla
 void nobHarborBuilding::Destroy()
 {
     em->RemoveEvent(orderware_ev);
-    players->getElement(player)->HarborDestroyed(this);
 
     // Der Wirtschaftsverwaltung Bescheid sagen
     GameClientPlayer& owner = gwg->GetPlayer(player);
     owner.RemoveWarehouse(this);
     owner.RemoveHarbor(this);
 
+    // Den Schiffen Bescheid sagen
+    players->getElement(player)->HarborDestroyed(this);
```

Comment:
Better use `owner.HarborDestroyed()`. I'd avoid the kinda global `player` variable (bad style and unsafe... it is the same as owner but that is not obvious)